feat(html): options and types

alexander-akait opened the pull request:
Options + types for HTML crate

alexander-akait commented on crates/html_node/src/lib.rs:
```diff
@@ -43,6 +49,93 @@ struct MinifyTask {
+    self_closing_void_elements: Option<bool>,
+}
+
+const fn true_by_default() -> bool {
```
Maybe we should export from `swc_html_minifier` to avoid copy/pasting, what do you think (same for belows)?

kdy1 replied:
I think copy/pasting is better

kdy1 replied:
But I think we should export config struct from `swc_html_minifier`

alexander-akait commented on packages/html/index.ts:
```diff
@@ -1,10 +1,47 @@
+  minifyJson?: boolean | { pretty?: boolean };
+  // TODO improve me after typing `@swc/css`
+  minifyJs?: boolean | { parser?: any; minifier?: any; codegen?: any };
+  minifyCss?: boolean | { parser?: any; minifier?: any; codegen?: any };
```
We will improve them late

alexander-akait commented on packages/html/index.ts:
```diff
@@ -1,10 +1,47 @@
+    | "conservative"
+    | "advanced-conservative"
+    | "only-metadata";
+  removeEmptyMetedataElements?: boolean;
```
Typo, I will fix them in `swc` itself and then fix it here

alexander-akait replied:
https://github.com/swc-project/swc/pull/5995

alexander-akait commented:
@kdy1 Rebased and fixed (the option name), anyway looks like we have the invalid version of `swc_html_utils` in `swc_html_minifier`

Because:

```
error[E0277]: the trait bound `std::string::String: Borrow<string_cache::atom::Atom<JsWordStaticSet>>` is not satisfied
   --> /home/akait/.cargo/registry/src/github.com-1ecc6299db9ec823/swc_html_minifier-0.75.25/src/lib.rs:373:75
    |
373 |         if let Some(global_pseudo_element) = HTML_ELEMENTS_AND_ATTRIBUTES.get(&js_word!("*")) {
    |                                                                           ^^^ the trait `Borrow<string_cache::atom::Atom<JsWordStaticSet>>` is not implemented for `std::string::String`
    |
    = help: the trait `Borrow<str>` is implemented for `std::string::String`

```

I have changed it to `JsWord` recently, so I think we need republish `swc_html_minifier` with fixed versions

kdy1 commented:
I'll publish a new version of `swc_html_utils` and dependants once https://github.com/swc-project/swc/pull/5998 (new version publish PR) is merged

kdy1 commented:
I published a new version with https://github.com/swc-project/swc/commit/b2da9f6c8f38f2f1f2b83388f5a98d79249a5916

alexander-akait commented:
Thank I will fix it soon

alexander-akait commented:
@kdy1 Ready for review + a new option added

kdy1 reviewed (approved):
Thanks!
